<a href="https://colab.research.google.com/github/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/Pre-Entrega_Maune%2C%20Juan_Federico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


<img  src="https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/Pre-Entrega%20IMG.PNG" width=100%>

#**MAUNE, Juan Federico**
#**Pre-Entrega de Proyecto**

**Bases de datos a utilizar:**
- [clientes.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/clientes.csv)
- [marketing.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/marketing.csv)
- [ventas.csv](https://github.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/blob/main/Pre-Entrega_Maune/datasets/ventas.csv)


#**Etapa 1: Recopilación y preparación de Datos**

##***Actividad 1: Carga de los sets de datos***

In [32]:
# se importa pandas
import pandas as pd

In [38]:
# se importa data set clientes
df_clientes = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/clientes.csv")

In [39]:
# se importa data set marketing
df_marketing = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/marketing.csv")

In [40]:
# se importa data set ventas
df_ventas = pd.read_csv("https://raw.githubusercontent.com/FedeM-Git/TT-2C2025-Data-Analitycs-Notebooks/refs/heads/main/Pre-Entrega_Maune/datasets/ventas.csv")

##***Actividad 2 y 3: Script Básico***

In [22]:
# se paso parte de los registros del set de datos de "Ventas" a una lista de diccionarios
lista_ventas = [
{'id_venta': 2824, 'producto': 'Tablet', 'precio': 74.09, 'cantidad': 5, 'fecha_venta': '45294', 'categoria': 'Electrónica'},
{'id_venta': 446, 'producto': 'Cortinas', 'precio': 62.10, 'cantidad': 4, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 630, 'producto': 'Adorno de pared', 'precio': 71.99, 'cantidad': 9, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 794, 'producto': 'Jarrón decorativo', 'precio': 96.97, 'cantidad': 7, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 882, 'producto': 'Cuadro decorativo', 'precio': 101.92, 'cantidad': 9, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 969, 'producto': 'Candelabro', 'precio': 90.70, 'cantidad': 12, 'fecha_venta': '45295', 'categoria': 'Decoración'},
{'id_venta': 1445, 'producto': 'Lavadora', 'precio': 28.41, 'cantidad': 10, 'fecha_venta': '45295', 'categoria': 'Electrodomésticos'},
{'id_venta': 1971, 'producto': 'Freidora eléctrica', 'precio': 122.25, 'cantidad': 11, 'fecha_venta': '45295', 'categoria': 'Electrodomésticos'},
{'id_venta': 2209, 'producto': 'Proyector', 'precio': 32.15, 'cantidad': 1, 'fecha_venta': '45295', 'categoria': 'Electrónica'},
{'id_venta': 256, 'producto': 'Cortinas', 'precio': 86.82, 'cantidad': 8, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 606, 'producto': 'Cortinas', 'precio': 106.66, 'cantidad': 11, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 704, 'producto': 'Jarrón decorativo', 'precio': 57.22, 'cantidad': 9, 'fecha_venta': '45296', 'categoria': 'Decoración'},
{'id_venta': 1362, 'producto': 'Heladera', 'precio': 109.49, 'cantidad': 11, 'fecha_venta': '45296', 'categoria': 'Electrodomésticos'},
{'id_venta': 2155, 'producto': 'Auriculares', 'precio': 84.80, 'cantidad': 10, 'fecha_venta': '45296', 'categoria': 'Electrónica'},
{'id_venta': 2947, 'producto': 'Cámara digital', 'precio': 107.78, 'cantidad': 3, 'fecha_venta': '45296', 'categoria': 'Electrónica'},
{'id_venta': 1495, 'producto': 'Lavadora', 'precio': 38.99, 'cantidad': 6, 'fecha_venta': '45297', 'categoria': 'Electrodomésticos'},
{'id_venta': 1572, 'producto': 'Cafetera', 'precio': 61.40, 'cantidad': 9, 'fecha_venta': '45297', 'categoria': 'Electrodomésticos'},
{'id_venta': 2180, 'producto': 'Consola de videojuegos', 'precio': 102.57, 'cantidad': 9, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2487, 'producto': 'Cámara digital', 'precio': 50.76, 'cantidad': 3, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2603, 'producto': 'Smartphone', 'precio': 87.05, 'cantidad': 5, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 2885, 'producto': 'Auriculares', 'precio': 100.40, 'cantidad': 6, 'fecha_venta': '45297', 'categoria': 'Electrónica'},
{'id_venta': 17, 'producto': 'Elementos de cerámica', 'precio': 108.48, 'cantidad': 11, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 183, 'producto': 'Espejo decorativo', 'precio': 82.76, 'cantidad': 8, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 560, 'producto': 'Adorno de pared', 'precio': 57.62, 'cantidad': 6, 'fecha_venta': '45298', 'categoria': 'Decoración'},
{'id_venta': 632, 'producto': 'Cuadro decorativo', 'precio': 30.49, 'cantidad': 12, 'fecha_venta': '45298', 'categoria': 'Decoración'}
]

In [23]:
# ventas acumuladas de la categoría "Decoración"
ventas_decoracion = sum( [venta ['precio']* venta['cantidad']
                          for venta in lista_ventas
                          if venta['categoria']=='Decoración'])
print(f'El monto obtenido por las ventas en la categoría Decoración es de $ {ventas_decoracion}')

El monto obtenido por las ventas en la categoría Decoración es de $ 8530.539999999999


In [24]:
# ventas acumuladas de la categoría "Electrodométicos"
ventas_electrodomesticos = sum( [venta ['precio']* venta['cantidad']
                                 for venta in lista_ventas
                                 if venta['categoria']=='Electrodomésticos'])
print(f'El monto obtenido por las ventas en la categoría Electrodomésticos es de $ {ventas_electrodomesticos}')

El monto obtenido por las ventas en la categoría Electrodomésticos es de $ 3619.7799999999997


In [25]:
# ventas acumuladas de la categoría "Electrónica"
ventas_electronica = sum( [venta ['precio']* venta['cantidad']
                           for venta in lista_ventas
                           if venta['categoria']=='Electrónica'])
print(f'El monto obtenido por las ventas en la categoría Electrónica es de $ {ventas_electronica}')

El monto obtenido por las ventas en la categoría Electrónica es de $ 3687.0


In [26]:
# con ayuda de AI realicé los mismos clculos pero en una sola sentencia
ventas_por_categoria = {
    categoria: sum(venta['precio'] * venta['cantidad']
                   for venta in lista_ventas if venta['categoria'] == categoria)
    for categoria in {venta['categoria'] for venta in lista_ventas}
}

print(ventas_por_categoria)

{'Electrónica': 3687.0, 'Decoración': 8530.539999999999, 'Electrodomésticos': 3619.7799999999997}


## ***Actividad 4: Calidad de los Datos***
**Análisis exploratorio con Pandas**

### ***DataFrame Clientes***

#### **Análisis Exploratorio**

In [65]:
df_clientes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 567 entries, 0 to 566
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   id_cliente  567 non-null    int64  
 1   nombre      567 non-null    object 
 2   edad        567 non-null    int64  
 3   ciudad      567 non-null    object 
 4   ingresos    567 non-null    float64
dtypes: float64(1), int64(2), object(2)
memory usage: 22.3+ KB


In [66]:
df_clientes.head(10)

id_cliente               nombre  edad                 ciudad  ingresos
0           1      Aloysia Screase    44          Mar del Plata  42294.68
1           2  Kristina Scaplehorn    25                Posadas  24735.04
2           3       Filip Castagne    50            Resistencia  35744.85
3           4          Liuka Luard    39           Bahía Blanca  27647.96
4           5        Dore Cockshtt    28                Rosario  28245.65
5           6        Patrick Earle    34  San Miguel de Tucumán  62763.31
6           7           Etan Deeth    35            Resistencia  37489.71
7           8       Booth Bielfelt    40                Córdoba  35255.94
8           9         Shirl Labone    29                Rosario  27592.08
9          10      Andy Mendenhall    52                Rosario  37153.94

In [67]:
df_clientes.describe()

id_cliente        edad      ingresos
count  567.000000  567.000000    567.000000
mean   284.000000   37.940035  34668.739012
std    163.823075   10.202885  12974.531446
min      1.000000   20.000000    170.290000
25%    142.500000   30.000000  26015.240000
50%    284.000000   37.000000  35066.830000
75%    425.500000   43.000000  42457.100000
max    567.000000   81.000000  88053.010000

In [68]:
df_clientes['ciudad'].value_counts()

ciudad
Mar del Plata            63
Rosario                  55
Posadas                  52
Resistencia              50
Córdoba                  49
Corrientes               47
Santa Fe                 46
Bahía Blanca             44
Salta                    43
Merlo                    43
San Miguel de Tucumán    39
Buenos Aires             36
Name: count, dtype: int64

In [69]:
df_clientes.isnull().sum()

id_cliente    0
nombre        0
edad          0
ciudad        0
ingresos      0
dtype: int64

In [70]:
df_clientes.duplicated().sum()

np.int64(0)

####**Observaciones**


- cantidad de registros: 567
- columnas: 5
- Precisión: los datos estan en los rangos esperados.
- Validez: los datos poseen los tipos de datos correctos.
- Completitud: no se observan registros con valores nulos.
- Cambios a realizar: no se obserban

### ***DataFrame Marketing***

#### **Análisis Exploratorio**

In [71]:
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id_campanha   90 non-null     int64  
 1   producto      90 non-null     object 
 2   canal         90 non-null     object 
 3   costo         90 non-null     float64
 4   fecha_inicio  90 non-null     object 
 5   fecha_fin     90 non-null     object 
dtypes: float64(1), int64(1), object(4)
memory usage: 4.3+ KB


In [72]:
df_marketing.head(10)

id_campanha             producto  canal  costo fecha_inicio   fecha_fin
0           74      Adorno de pared     TV   4.81   20/03/2024  03/05/2024
1           12               Tablet   RRSS   3.40   26/03/2024  13/05/2024
2           32      Lámpara de mesa  Email   5.54   28/03/2024  20/04/2024
3           21           Smartphone   RRSS   6.37   29/03/2024  16/05/2024
4           58             Alfombra  Email   4.25   31/03/2024  05/05/2024
5           85           SmartWatch     TV   5.07   01/04/2024  05/05/2024
6           36     Plancha de vapor  Email   5.41   02/04/2024  01/06/2024
7           57             Batidora  Email   4.48   10/04/2024  08/06/2024
8           44      Adorno de pared  Email   5.08   13/04/2024  10/05/2024
9           84  Parlantes Bluetooth     TV   4.42   17/04/2024  05/05/2024

####**Observaciones**

# ejercicios

- Etapa 1
    4. Introducción a Pandas: realizar un análisis exploratorio inicial de los DataFrames.

    5. Calidad de Datos: Identificar valores nulos y duplicados en los conjuntos de datos. Documentar el estado inicial de los datos.

- Etapa 2
    1. Limpieza de Datos: Limpiar el conjunto de datos eliminando duplicados y caracteres no deseados. Documentar el proceso y los resultados.

    2. Transformación de Datos: Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento.

    3. Agregación: Resumir las ventas por categoría de producto y analizar los ingresos generados.

    4. Integración de Datos: Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.